### PR TITLE
Prepare for 0.12 release

### PR DESCRIPTION
**Pull request by trbromley**
This will not pass tests yet

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
         run: |
           cd main
           python -m pip install --upgrade pip
-          pip install git+https://github.com/PennyLaneAI/pennylane.git
+          pip install pennylane --upgrade
```
Comment (antalszava) [inline .github/workflows/build.yml:59]: This is redundant and would need to be removed such that we pull PL from `requirements.txt` (https://github.com/PennyLaneAI/pennylane-lightning/pull/43#discussion_r505728317)
```suggestion
```

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -71,7 +71,7 @@ jobs:
       - name: Install cibuildwheel
         run: |
           python -m pip install --upgrade pip
-          pip install git+https://github.com/PennyLaneAI/pennylane.git
+          pip install pennylane --upgrade
```
Comment (antalszava) [inline .github/workflows/wheels.yml:74]: Similarly, can be removed (installing the wheel will pull in PL)
```suggestion
```

File: pennylane_lightning/lightning_qubit.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ class LightningQubit(DefaultQubit):
 
     name = "Lightning Qubit PennyLane plugin"
     short_name = "lightning.qubit"
-    pennylane_requires = ">=0.11"
+    pennylane_requires = ">=0.12"
```
Comment (antalszava) [inline pennylane_lightning/lightning_qubit.py:56]: Nice! :100:

**Review by antalszava — APPROVED** [review]
Looks good, thank you! :blush:

**Comment by trbromley** [comment]
Thanks @antalszava!

**Pull request merged**